layers/+chat/circe/packages.el

```diff
@@ -0,0 +1,162 @@
+
+      (add-hook 'circe-chat-mode-hook #'circe/set-prompt)
+
+      ;; Timestamps in margins.
+
+      (setq lui-time-stamp-position 'right-margin)
+      (setq lui-time-stamp-format "%H:%M")
+      (setq lui-fill-type nil)))
+
+  (use-package circe-color-nicks
+    :commands enable-circe-color-nicks
+    :init
+    (with-eval-after-load 'circe
+      (enable-circe-color-nicks))
+    :config
+    (setq circe-color-nicks-everywhere t))
+
+  (use-package lui
+    :init
+    (progn
+      (defun circe/set-local-vars ()
+        (setq fringes-outside-margins t)
+        (setq right-margin-width 5)
+        (setq word-wrap t)
+        (setq wrap-prefix "    "))
+
+      (add-hook 'lui-mode-hook #'circe/set-local-vars)))
+
+  (use-package lui-autopaste
+    :commands enable-lui-autopaste
+    :init
+    (add-hook 'circe-channel-mode-hook #'enable-lui-autopaste)))
+
+(defun circe/post-init-helm ()
+  (use-package helm
+    :defer t
+    :config
+    (dolist (mode circe-modes)
+      (add-to-list 'helm-mode-no-completion-in-region-in-modes mode))))
+
+(defun circe/init-helm-circe ()
+  (use-package helm-circe
+    :defer t
+    :after circe
+    :config
+    ;; Helm keybindings
+    (dolist (mode circe-modes)
+      (spacemacs/declare-prefix-for-mode mode "mh" "helm")
+      (spacemacs/set-leader-keys-for-major-mode mode
+        "h h" 'helm-circe
+        "h s" 'helm-circe-servers
+        "h q" 'helm-circe-queries
+        "h b" 'helm-circe-by-server
+        "h n" 'helm-circe-new-activity
+        ))))
+
+(defun circe/init-circe-notifications ()
+  (use-package circe-notifications
+    :commands enable-circe-notifications
+    :after circe
+    :init
+    (add-hook 'circe-server-connected-hook #'enable-circe-notifications)
+    :config
+    (progn
+      ;;; HACK: Fix broken implementation of internal function.
+
+      (defun circe-notifications-nicks-on-all-networks ()
+        "Get a list of all nicks in use according to `circe-network-options'."
+        (delete-dups (mapcar (lambda (opt)
+                               (plist-get (cdr opt) :nick))
+                             circe-network-options)))
+
+      (when (eq system-type 'darwin)
+        (setq circe-notifications-alert-style 'osx-notifier)))))
+
+(defun circe/post-init-persp-mode ()
+  ;; do not save circe buffers
+  (with-eval-after-load 'persp-mode
+    (push (lambda (b) (with-current-buffer b (eq major-mode 'circe-mode)))
+          persp-filter-save-buffers-functions))
+
+
+
+   (spacemacs|define-custom-layout circe-spacemacs-layout-name
+     :binding circe-spacemacs-layout-binding
+     :body
+     (progn
+       (defun spacemacs-layouts/add-circe-buffer-to-persp ()
+         (persp-add-buffer (current-buffer)
+                           (persp-get-by-name
+                            circe-spacemacs-layout-name)
+                           ))
+       (add-hook 'circe-mode-hook #'spacemacs-layouts/add-circe-buffer-to-persp)
+       (call-interactively 'circe)))
+
+  )
```

Comment:
the extra whitespace and hanging parenthesis in `circe/post-init-persp-mode` is considered bad form.